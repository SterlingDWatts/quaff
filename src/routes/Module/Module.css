.Module {
  padding: 16px;
  max-width: 944px;
  margin: 0 auto;
}

.Module header {
  font-size: 60px;
  line-height: 1;
  font-weight: 600;
  margin: 65px 0;
}

.Module h2 {
  font-size: 24px;
  line-height: 1;
  font-weight: 600;
  margin: 30px 0;
}

.Module h3 {
  font-size: 20px;
  line-height: 1;
  font-weight: 600;
  margin-bottom: 10px;
}

.Module h4 {
  font-size: 17px;
  line-height: 1;
  font-weight: 600;
  font-style: italic;
  margin-bottom: 10px;
}

.Module p {
  margin-bottom: 20px;
  line-height: 1.7;
  text-indent: 24px;
}

.Module em {
  font-weight: 600;
}

.Module sub {
  vertical-align: sub;
  font-size: 10px;
}

.Module img {
  width: 100%;
}

.Module__buttons {
  display: flex;
  justify-content: center;
}
