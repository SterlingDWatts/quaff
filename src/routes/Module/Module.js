import React, { Component } from "react";
import { Route, Link } from "react-router-dom";
import { Button } from "../../components/Utils/Utils";
import ModA from "../../components/ModuleContents/ModA";
import ModB from "../../components/ModuleContents/ModB";
import "./Module.css";

class Module extends Component {
  static defaultProps = {
    match: {
      params: {
        moduleId: -1,
      },
    },
  };
  renderModule = () => {
    return (
      <div>
        <Route path="/learn/1" component={ModA} />
        <Route path="/learn/2" component={ModB} />
        <Route path="/learn/3" component={ModB} />
        <div className="Module__buttons">
          <Link to={`/test/${this.props.match.params.moduleId}`}>
            <Button className="form__button">Test</Button>
          </Link>
        </div>
      </div>
    );
  };

  render() {
    return <div className="Module">{this.renderModule()}</div>;
  }
}

export default Module;
