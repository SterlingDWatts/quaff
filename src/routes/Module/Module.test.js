import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Module from "./Module";

describe("Module Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Module />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Module />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
