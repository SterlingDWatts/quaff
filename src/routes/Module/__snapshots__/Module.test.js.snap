// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Component renders the UI as expected 1`] = `
<div
  className="Module"
>
  <div>
    <div
      className="Module__buttons"
    >
      <a
        href="/test/-1"
        onClick={[Function]}
      >
        <button
          className="Button form__button"
        >
          Test
        </button>
      </a>
    </div>
  </div>
</div>
`;
